subiquity/server/controllers/filesystem.py

```diff
@@ -108,36 +113,122 @@ async def apply_autoinstall_config(self, context=None):
                 "autoinstall config did not create needed bootloader "
                 "partition")
 
-    async def GET(self, wait: bool = False) -> StorageResponse:
+    def guided_direct(self, disk):
+        self.reformat(disk)
+        result = {
+            "size": disk.free_for_partitions,
+            "fstype": "ext4",
+            "mount": "/",
+            }
+        self.partition_disk_handler(disk, None, result)
+
+    def guided_lvm(self, disk, lvm_options=None):
+        self.reformat(disk)
+        if DeviceAction.TOGGLE_BOOT in disk.supported_actions:
+            self.add_boot_disk(disk)
+        self.create_partition(
+            device=disk, spec=dict(
+                size=dehumanize_size('1G'),
+                fstype="ext4",
+                mount='/boot'
+                ))
+        part = self.create_partition(
+            device=disk, spec=dict(
+                size=disk.free_for_partitions,
+                fstype=None,
+                ))
+        spec = dict(name="ubuntu-vg", devices=set([part]))
+        if lvm_options and lvm_options['encrypt']:
+            spec['password'] = lvm_options['luks_options']['password']
+        vg = self.create_volgroup(spec)
+        # There's no point using LVM and unconditionally filling the
+        # VG with a single LV, but we should use more of a smaller
+        # disk to avoid the user running into out of space errors
+        # earlier than they probably expect to.
+        if vg.size < 10 * (2 << 30):
```

Comment:
And we have some sizes here too.  I suggest pulling all the sizes into one spot so that we can concisely set policies of minimum/use all/use half/etc.